Frends.AzureDataLake.DeleteDirectory/Frends.AzureDataLake.DeleteDirectory/Frends.AzureDataLake.DeleteDirectory.cs

```diff
@@ -0,0 +1,79 @@
+﻿namespace Frends.AzureDataLake.DeleteDirectory;
+
+using System;
+using System.ComponentModel;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Azure.Identity;
+using Azure.Storage.Files.DataLake;
+using Frends.AzureDataLake.DeleteDirectory.Definitions;
+
+/// <summary>
+/// Main class of the Task.
+/// </summary>
+public static class AzureDataLake
+{
+    /// <summary>
+    /// Frends Task for deleteing a directory in Azure Data Lake container.
+    /// [Documentation](https://tasks.frends.com/tasks/frends-tasks/Frends.AzureDataLake.DeleteDirectory)
+    /// </summary>
+    /// <param name="input">Input parameters.</param>
+    /// <param name="cancellationToken">Token generated by Frends to stop this task.</param>
+    /// <returns>Object { string ContainerWasDeleted, string Message }</returns>
+    public static async Task<Result> DeleteDirectory([PropertyTab] Input input, [PropertyTab] Options options, CancellationToken cancellationToken)
+    {
+        if (input.ConnectionMethod is ConnectionMethod.OAuth2 && (input.ApplicationID is null || input.ClientSecret is null || input.TenantID is null || input.StorageAccountName is null))
+            throw new ArgumentNullException("Input.StorageAccountName, Input.ClientSecret, Input.ApplicationID and Input.TenantID parameters can't be empty when Input.ConnectionMethod = OAuth.");
+        if (string.IsNullOrWhiteSpace(input.ConnectionString) && input.ConnectionMethod is ConnectionMethod.ConnectionString)
+            throw new ArgumentNullException("ConnectionString parameter can't be empty when Input.ConnectionMethod = ConnectionString.");
+        if (string.IsNullOrWhiteSpace(input.ContainerName))
+            throw new ArgumentNullException("ContainerName parameter can't be empty.");
+        if (string.IsNullOrWhiteSpace(input.DirectoryName))
+            throw new ArgumentNullException("DirectoryName parameter can't be empty.");
+
+        try
+        {
+            DataLakeDirectoryClient directory = GetDataLakeDirectory(input);
+
+            if (!await directory.ExistsAsync(cancellationToken) && !options.ThrowErrorIfDirectoryDoesNotExists)
```

Comment:
Save ExistsAsync call result to a variable and then do a simple if/else on ThrowErrorIfDirectoryDoesNotExists.